### ISSUE TITLE
macOS "Error: Failed to fetch backend nodes."

### ISSUE THREAD

**Issue by hopelesscase** [chaiNNer-org/chaiNNer#2223]
**Information:**

- Chainner version: 20.1
- OS: macOS 13.5.2

**Description**
Installed (i.e. copied) the latest version and got this...
```
Critical Error

A critical error occurred while processing the node data returned by the backend.

Error: Failed to fetch backend nodes.
```
No matter how long I wait it never budges. I only got that error when I went to get the version info. I click to exit the application, but it still shows the little loading window. The only wait to close it is by force quitting.

**Logs**
Skipped since I don't really want to risk putting my user name out there. If they are crucial I'll edit them, but I'd rather not risk it if I can...

Privacy information: Log files contain file paths of Chainner's application files and the files you processed with Chainner. These file paths may contain the name of your account (e.g. Users/michael/...). If your account is your real name, and you are not comfortable sharing, then do not upload your logs or replace all identifying text with something like "fake name".

**Comment by stonerl** [comment]
When does this bug message appear? Right after startup?

The main.log would be helpful. You can delete the old log and start chaiNNer. The new log will contain only a couple of lines.

Open the log in a texteditor and remove your username.

**Comment by stonerl** [comment]
@RunDevelopment I think this is related to #2038

**Comment by hopelesscase** [comment]
> When does this bug message appear? Right after startup?
> 
> The main.log would be helpful. You can delete the old log and start chaiNNer. The new log will contain only a couple of lines.
> 
> Open the log in a texteditor and remove your username.

[chaiNNer_Logs.zip](https://github.com/chaiNNer-org/chaiNNer/files/12643240/chaiNNer_Logs.zip)

😣 I hope I got all cases of my username… I also hope this helps shed some light on it.

**Comment by hopelesscase** [comment]
I'm sorry, I forgot to mention that it occurs immediately at startup. I open the app and then I can't get past the smaller loading window. Using the newest version I've never gotten past the smaller loading window.

When I downgrade back to 19.4 everything seems to work okay. I don't understand if or how to apply a workaround nor if it's something like a file I should trash before using the newer version.

Sorry for forgetting to mention some stuff. Thank you for the great app and I hope I didn't break something with my computer's copy. 😅

**Comment by joeyballentine** [comment]
The reason it's erroring is because it's trying to access a file from your system 3.9 python. I'm not sure why it would be doing this, so I'm not sure what to do to fix it. But it should never be doing that

**Comment by stonerl** [comment]
@hopelesscase Where did you get the python 3.9 libraries, you installed manually, from?

**Comment by hopelesscase** [comment]
I downloaded my python libraries via Homebrew, but I set it so when I enter "python3" in the Terminal it shows python 3.11.5. I technically have three different versions of python because different formulae depend on different ones.

However, thanks to the insight that chaiNNer was trying to use my system python I got a workaround to get it running.

1. Downgrade to 19.4
2. Go into Settings and disable the option to use system python over the included version
3. Upgrade to newest chaiNNer and re-download dependencies

After disabling the preference of my system python over the app's python _before_ upgrading I got it to work.

Thank you!
